test/verify/gemm_softmax_gemm_relu.cpp

```diff
@@ -41,17 +42,27 @@ struct gemm_softmax_gemm_relu : verify_program<gemm_softmax_gemm_relu>
         auto b1          = mm->add_parameter("3", m1_shape);
         std::vector<float> eights(m2_elements, 0.125);
         auto eight = mm->add_literal(migraphx::literal{m2_shape, eights});
-        std::vector<float> zeros(m2_elements, 0);
-        auto zero = mm->add_literal(migraphx::literal{m2_shape, zeros});
 
         b = mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {0, 1, 3, 2}}}), b);
-        auto gemm1   = mm->add_instruction(migraphx::make_op("dot"), a, b);
-        auto scale   = mm->add_instruction(migraphx::make_op("mul"), gemm1, eight);
-        auto bias    = mm->add_instruction(migraphx::make_op("add"), scale, zero);
-        auto softmax = mm->add_instruction(migraphx::make_op("softmax", {{"axis", 3}}), bias);
+        auto gemm1 = mm->add_instruction(migraphx::make_op("dot"), a, b);
+        auto scale = mm->add_instruction(migraphx::make_op("mul"), gemm1, eight);
+
+        std::optional<migraphx::instruction_ref> add_bias{std::nullopt};
+        if constexpr(WithBias)
+        {
+            std::vector<float> one_tenth(m2_elements, 0.1);
+            auto bias = mm->add_literal(migraphx::literal{m2_shape, one_tenth});
```

Comment:
Can you try making this bias a non-standard shaped  as a test ?

Reply:
Hi @umangyadav, thanks for pointing this out. I forgot to change this part after debugging during the development. Now this part looks like the following

https://github.com/ROCm/AMDMIGraphX/blob/da734c677ff94d797468f232edc0295886769dd7/test/verify/gemm_softmax_gemm_relu.cpp#L52-L55

@manupak and I wanted the bias term to be a function parameter. We think it reflects better what we wanted to achieve

Reply:
@umangyadav

Here is the test with non-standard shape of the bias term.

https://github.com/ROCm/AMDMIGraphX/blob/6720325cfecf678473780e222bb3a1482b73fac5/test/verify/gemm_softmax_gemm_relu.cpp#L54-L59

 Everything works